### ISSUE TITLE
strange code in LevelSolver.cs

### ISSUE THREAD

**Issue by izatar** [Microsoft/HoloToolkit-Unity#238]
Here is the InitializeSolver() function I see in my sync'd holotoolkit-unity:

```

    public bool InitializeSolver()
    {
        if (IsSolverInitialized ||
            !SpatialUnderstanding.Instance.AllowSpatialUnderstanding)
        {
            return IsSolverInitialized;
        }

        if (SpatialUnderstandingDllObjectPlacement.Solver_Init() > 1)
        {
            IsSolverInitialized = true;
        }
        return IsSolverInitialized;
    }

```

I see Solver_Init() resolving to:

```

#if UNITY_METRO && !UNITY_EDITOR
        [DllImport("SpatialUnderstanding")]
        public static extern int Solver_Init();
#else
```

which resolves to something in the C++ code in Holotoolkit:

```
EXTERN_C __declspec(dllexport) int Solver_Init()
{
    FRGSolver_W& solver = UnderstandingMgr_W::GetUnderstandingMgr().GetSolver();

    // Init the solver
    solver.Init(UnderstandingMgr_W::GetUnderstandingMgr().GetPlayspaceInfosPtr(),
        &UnderstandingMgr_W::GetUnderstandingMgr().GetPlayspaceInfos().m_TopologyAnalyzer,
        &UnderstandingMgr_W::GetUnderstandingMgr().GetPlayspaceInfos().m_ShapeAnalyzer);

    return 1;
}

```

It appears the Solver_Init() always returns 1.  But, the function in LevelSolver.cs  is checking for ">1" so it appears that " IsSolverInitialized = true;" never happens.  I put in a debug there and it does in fact look like it never gets set right, and so appears to be  calling InitializeSolver() on every update.

**Comment by NeerajW** [comment]
@jevertt looks like the check should be updated to == 1 instead of > 1?

**Comment by jevertt** [comment]
It looks like solver.Init() can be called multiple times without any bad side-effects, so this wasn't noticed. But, yes - that's not intentional - it's a bug.

**Comment by keveleigh** [comment]
Fixed with #371
